scripts/pylint_custom_plugin/pylint_guidelines_checker.py

```diff
@@ -886,17 +887,30 @@ def visit_functiondef(self, node):
         """
         try:
             if node.parent.name.endswith("Client") and node.parent.name not in self.ignore_clients and node.is_method():
+                
+
+                # TODO: Flag functions that return an ItemPaged but don't start with "list"
+
                 if node.name.startswith("list"):
-                    try:
-                        # infer_call_result gives the method return value as a string
-                        returns = next(node.infer_call_result()).as_string()
-                        if returns.find("ItemPaged") == -1 and returns.find("AsyncItemPaged") == -1:
-                            self.add_message(
-                                msgid="client-list-methods-use-paging", node=node, confidence=None
-                            )
-                    except (astroid.exceptions.InferenceError, AttributeError): # astroid can't always infer the return
-                        logger.debug("Pylint custom checker failed to check if client list method uses core paging.")
-                        pass
+                    paging_class = False
+
+                    for inner_node in node.body:
+                        # If this is a return node
+                        if isinstance(inner_node, astroid.Return):
+                            try:
+                                # Checking that what we are returning is a function call
+                                if isinstance(inner_node.value, astroid.Call):
```

Comment:
does this work for either an ItemPaged or a function call which returns an ItemPaged? If the return type is very incorrect, like returning `None` or something, will that get caught?

Reply:
Good point, I will test with those!